src/intrprtr.c

```diff
@@ -4583,19 +4583,28 @@ void            IntrInfoBegin( void )
 }
 
 static Obj InfoDecision;
-static Obj InfoDecisionFast;
 static Obj IsInfoClassListRep;
 
 Obj InfoCheckLevel(Obj selectors, Obj level)
 {
-    if (IS_POS_INTOBJ(level) &&
-        CALL_1ARGS(IsInfoClassListRep, selectors) == True) {
+    // Fast-pass the most common failing case
+    // The fast-pass never returns True because the GAP function
+    // InfoDecision sets a number of variables as well as returning True
```

Comment:
This explanation feels a bit backwards to me (but that may just me being dense). I'd suggest something like this: 
> The fast-path only deals with the case were False is returned, as in the
True case, `InfoData.LastClass` and `InfoData.LastLevel` need to be set.


But actually, I think we can get rid of this exception, by getting rid of `InfoData`. In the meantime, though, of course it makes sense to document this exception.